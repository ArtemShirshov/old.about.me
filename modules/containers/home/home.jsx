// @flow
import React, {PureComponent, type Node, Fragment} from 'react';
import {connect} from 'react-redux';

import {fetchPostData} from 'reducers/post/post';
import {RecordFormConnected} from 'components/recordForm/recordForm';

import s from './home.scss';

type PropsType = {};

/**
 * Main container for home page
 *
 * @return {Node} - Element
 */
export class Home extends PureComponent<PropsType> {
  componentDidMount() {
    this.props.fetchPostData();
  }

  /**
   * Render jsx to html
   *
   * @returns {Node} Rendered react component
   */
  render(): Node {
    return (
      <div className={s.wrap}>
        <RecordFormConnected />
      </div>
    );
  }
}

export const mapStateToProps = (store: ApplicationStoreType) => ({});

export const mapDispatchToProps = {
  fetchPostData,
};

export const HomeConnected = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Home);
