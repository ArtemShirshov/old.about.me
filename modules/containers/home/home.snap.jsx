import React from 'react';
import renderer from 'react-test-renderer';

import {Home} from './home';

jest.mock('components/__helpers/popup/popup', () => ({Popup: 'Popup'}));
jest.mock('components/__helpers/header/header', () => ({
  HeaderConnected: 'HeaderConnected',
}));
jest.mock('components/__helpers/footer/footer', () => ({Footer: 'Footer'}));
jest.mock('components/home/topBanner/topBanner', () => ({
  TopBanner: 'TopBanner',
}));
jest.mock('components/home/brandsStrip/brandsStrip', () => ({
  BrandsStrip: 'BrandsStrip',
}));
jest.mock('components/home/youCovered/youCovered', () => ({
  YouCovered: 'YouCovered',
}));
jest.mock('components/home/lenses/lenses', () => ({Lenses: 'Lenses'}));
jest.mock('components/home/mirror/mirror', () => ({Mirror: 'Mirror'}));
jest.mock('components/home/offers/offers', () => ({Offers: 'Offers'}));
jest.mock('components/home/getPlans/getPlans', () => ({GetPlans: 'GetPlans'}));
jest.mock('components/home/brandsGallery/brandsGallery', () => ({
  BrandsGallery: 'BrandsGallery',
}));

const props = {
  cmsPopup: {
    visibility: true,
    content: 'testContent',
    params: {},
  },
  getCustomer: () => {},
};

describe('<Home/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Home {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with html content for popup', () => {
    props.cmsPopup.content = {html: '<p>some html content</p>'};
    const tree = renderer.create(<Home {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
