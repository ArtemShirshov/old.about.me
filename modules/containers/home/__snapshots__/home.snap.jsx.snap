// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> renders correctly 1`] = `
Array [
  <HeaderConnected />,
  <TopBanner />,
  <BrandsStrip />,
  <YouCovered />,
  <Lenses />,
  <Mirror />,
  <Offers />,
  <BrandsGallery />,
  <GetPlans />,
  <Footer />,
  <Popup
    showClose={true}
    visible={true}
  >
    testContent
  </Popup>,
]
`;

exports[`<Home/> renders correctly with html content for popup 1`] = `
Array [
  <HeaderConnected />,
  <TopBanner />,
  <BrandsStrip />,
  <YouCovered />,
  <Lenses />,
  <Mirror />,
  <Offers />,
  <BrandsGallery />,
  <GetPlans />,
  <Footer />,
  <Popup
    showClose={true}
    visible={true}
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>some html content</p>",
        }
      }
    />
  </Popup>,
]
`;
